### DISCUSSION TITLE
Warning: If you intend to use initial guesses, you must use a fifth output as [sol,problem,~,~,P] = P(p)

### DISCUSSION THREAD

**Discussion by mrkento11** [yalmip/YALMIP#1236] [Problems using YALMIP] [answered]
Dear Forum and dear Professor Löfberg,
Thank you for your continuous help.

Today I want to ask why am I getting the warning: "Warning: If you intend to use initial guesses, you must use a fifth output as [sol,problem,~,~,P] = P(p)" and how to potentially fix it.

I am optimizing several times a mathematical problem and changing just a few parameters in it at every iteration, so I am using an optimizer to do this. As explained in this thread at [Google Groups](https://groups.google.com/g/yalmip/c/cQqtbEVVk_Y), I can indeed "turn on warm start but [I] will not be able to pick the initial guess [...]". Given that my problem is so similar, and want to save some time searching through the BnB tree, I added in my optimizer settings the option for warm start (...,'usex0',1,...).

However, I am getting the warning mentioned there. My interpretation is that I should write:
[problem_solution,errorcode,~,~,Initial_guess] = optimizer_object({var1,var2}, Initial guess). But I am very uncertain about this reasoning.

Thank you in advance for your time.

**Comment by johanlofberg** [comment, marked as answer]
With optimizer object, the call will not be warm-started the first time you use it (i.e. any assignment using assign is redundant), but the idea is that when optimizer makes a call to its solver, the solution is saved, and reused. However, for it to be saved, it has to output it self from the call, with the warm-start info appended.

**Reply by mrkento11** [reply]
Thank you Professor.

In case somebody goes back to this thread, the syntax I used was:
`[sol, problem,~,~,optimizer_object] = optimizer_object({Parameters});`

This little change made my code run in a little less than half of the time (~50%+ efficiency increase when measuring time).

**Reply by johanlofberg** [reply]
You don't have to put the parameter in a cell (unless you mean you created a cell list in the code somewhere before or something)

**Reply by mrkento11** [reply]
Yes, Professor. In fact, that is the case: when I created my optimizer object, I define the optimizer object to be a function of more than one parameter, thus I created a cell list:

`optimizer_object = optimizer(Constraints, Cost, options,{Parameter1,Parameter2},{var1,var2,var3,Cost});`

So, when I actually use the optimizer object, I call the parameters using a cell list:
`[sol, problem,~,~,optimizer_object] = optimizer_object({Parameters1,Parameter2});`

Thanks for the comment.

**Reply by johanlofberg** [reply]
ok (`optimizer_object(Parameters1,Parameter2)` works just as well)
